pylint/checkers/classes/class_checker.py

```diff
@@ -2061,9 +2061,6 @@ def _check_init(self, node: nodes.FunctionDef, klass_node: nodes.ClassDef) -> No
 
             if decorated_with(node, ["typing.overload"]):
                 continue
```

Comment:
The lines below are no longer necessary now that we skip `object.__init__` because of its abstractness.